sys/include/net/gnrc/netif/dedup.h

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_netif_dedup    Link-layer Broadcast deduplication
+ * @ingroup     net_gnrc_netif
+ * @brief       Deduplicates broadcast link-layer packets best-effort style
+ *
+ * To activate, use `USEMODULE += gnrc_netif_dedup` in your applications
+ * Makefile. Also make sure the link-layer you use supports the module.
+ * Currently supported are
+ *
+ * - IEEE 802.15.4
+ *
+ * @{
+ *
+ * @file
+ * @brief
+ *
+ * @author  Martine Lenders <m.lenders@fu-berlin.de>
+ */
+#ifndef NET_GNRC_NETIF_DEDUP_H
+#define NET_GNRC_NETIF_DEDUP_H
+
+#include <stdint.h>
+
+#include "net/gnrc/netif/conf.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Structure to store information on the last broadcast packet received
+ */
+typedef struct {
+    uint8_t src[GNRC_NETIF_L2ADDR_MAXLEN];  /**< link-layer source address */
+    uint16_t seq;                           /**< link-layer sequence number */
```

Comment:
FYI I chose `uint16_t` instead of `uint8_t` since this implementation supposed to be independent of IEEE 802.15.4. I'm not aware of any packet switching Link-Layer with packets `>UINT16_MAX` so I figured `uint16_t` might be enough.